src/DesktopIntegration.vala

```diff
@@ -43,6 +48,17 @@ public class Gala.DesktopIntegration : GLib.Object {
         // TODO: figure out if there's a better way to handle ws rearrangement
         display.window_created.connect ((window) => {
             window.workspace_changed.connect (() => windows_changed ());
+
+            time_appeared_on_workspace[window] = GLib.get_real_time ();
```

Comment:
I think we'd rather want to use [get_monotonic_time](https://valadoc.org/glib-2.0/GLib.get_monotonic_time.html) here?